### PR TITLE
NODE-2002 Initial project structure

### PR DESCRIPTION
**Pull request by michaelgoin**
* Setups up initial project structure and scripts using skeleton of spanBroker and test.

I think we'll still eventually need to define typings in package.json to allow grabbing our types but might as well do that as a part of our first API's being available. 

## Linting notes
Leverages recommended settings from `@typescript-eslint`. These may require further tweaking to get linting just right. https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#supported-rules. ESLint seems to be the current focused direction for TypeScript right now. Leveraging the TypeScript plugin was the path of least resistance thus far.

## Tap VS TypeScript
Right now Tap doesn't publish types, so we have our own type definition. It's mostly based on: https://github.com/tapjs/node-tap/pull/256/files with some tweaks for our standards and adding `autoend`, etc. Not sure what sort of attribution we might need for that? Downside is we'll need to keep in sync with updates and may be other awkwardness. So we'll see if we get forced to punt on TS or use a different framework. isaacs doesn't really want to land types w/o in depth testing to make sure they stay in sync.

### CODE REVIEW

**Comment by NatalieWolfe** [comment]
Re: tap types, [tape](https://www.npmjs.com/package/tape) has types. https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/tape

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    "sourceType": "module"
+  },
+  "rules": {
+    "indent": ["off", 2, {"SwitchCase": 1}],
+    "@typescript-eslint/indent": ["error", 2, {"SwitchCase": 1}],
+    "@typescript-eslint/member-delimiter-style": [
+      "error",
+      {
+        "multiline": {
+            "delimiter": "none"
```
Comment (NatalieWolfe) [inline .eslintrc]: I like how our linter config wouldn't pass linting. :)

**Comment by michaelgoin** [comment]
@NatalieWolfe i tried tape briefly and it had a slightly different API, so i punted on it. i'm not against it but the start of our tests would just have to be a little different from usual. not sure what other differences may be.

**Comment by lykkin** [comment]
i'd rather put up with some api differences than have to manually keep the type def up to date.

File: package.json [hunk 2]
```diff
@@ -4,7 +4,11 @@
   "description": "Open Tracing API for New Relic.",
   "main": "index.js",
   "scripts": {
-    "test": "tap tests/**/*.tap.js"
+    "build": "tsc",
+    "lint": "./node_modules/.bin/eslint 'src/**' 'tests/**'",
+    "prepare": "npm run build",
+    "unit": "./node_modules/.bin/ts-node ./node_modules/.bin/tape tests/unit/**/*.tap.ts",
```
Comment (NatalieWolfe) [inline package.json]: Too late now, but `node_modules/.bin/` is unnecessary in npm scripts. npm adds that directory to `PATH` when it executes these.

**Pull request merged**